src/core/prompts/sections/custom-instructions.ts

```diff
@@ -226,16 +226,20 @@ export async function addCustomInstructions(
 		)
 	}
 
-	// Add global instructions first
-	if (typeof globalCustomInstructions === "string" && globalCustomInstructions.trim()) {
-		sections.push(`Global Instructions:\n${globalCustomInstructions.trim()}`)
-	}
-
-	// Add mode-specific instructions after
+	// Add mode-specific instructions first (higher priority)
 	if (typeof modeCustomInstructions === "string" && modeCustomInstructions.trim()) {
 		sections.push(`Mode-specific Instructions:\n${modeCustomInstructions.trim()}`)
 	}
 
+	// Only add global instructions if they differ from mode-specific instructions
```

Comment:
I think I don't quite understand what's going on here - is there code somewhere that defaults the global instructions to the mode-specific instructions if not present?

Reply:
here: https://github.com/RooVetGit/Roo-Code/blob/8447670499dc07b838a7cf659c4d1bd834e7a642/src/core/prompts/system.ts#L148

I wanted to change as little as possible to the code logic. Hence, the performed change for a correct api post without duplicate text.

Did you get the same result as described in the api post using postman or another proxy?

Reply:
Let me know what you think of https://github.com/RooVetGit/Roo-Code/pull/3638 - I think it might fix the root cause.